FlashCap.Core/Internal/AVFoundation/AVCaptureSession.cs

```diff
@@ -18,66 +18,173 @@ partial class LibAVFoundation
-                LibObjC.GetSelector("addInput:"),
-                input.Handle);
 
         public void AddOutput(AVCaptureOutput output)
         {
-            IntPtr allocSel = LibObjC.GetSelector("alloc");
-            IntPtr initSel = LibObjC.GetSelector("init");
-            
-            var videoDataOutputObj = output as AVCaptureVideoDataOutput ;
+            try
+            {
+                ValidateHandle();
+                
+                _videoDataOutput = output as AVCaptureVideoDataOutput ;
 
-            if (videoDataOutputObj == null)
+                if (_videoDataOutput == null)
+                {
+                    throw new Exception("Failed to get video data output");
+                }
+                
+                var videoDataOutput = _videoDataOutput.Handle;
+                
+                LibObjC.SendNoResult(
+                    Handle,
+                    LibObjC.GetSelector("addOutput:"),
+                    videoDataOutput);
+                
+            } catch (Exception ex)
             {
-                throw new Exception("Failed to get video data output");
+                Console.WriteLine($"Error calling addOutput: {ex.Message}");
+                throw;
+            }
+        }
+
+        public bool CanAddOutput(AVCaptureOutput output)
+        {
+            try
+            {
+                ValidateHandle();
+                return LibObjC.SendAndGetBool(
+                    Handle,
+                    LibObjC.GetSelector("canAddOutput:"),
+                    output.Handle);
+            } catch (Exception ex)
+            {
+                Console.WriteLine($"Error calling canAddOutput: {ex.Message}");
+                throw;
+            }
+        }
+
+
+        public void StartRunning()
+        {
+            try
+            {
+                ValidateHandle();
+                LibObjC.SendNoResult(
+                    Handle,
+                    LibObjC.GetSelector("startRunning"));
+            } catch (Exception ex)
+            {
+                Console.WriteLine($"Error starting AVCaptureSession: {ex.Message}");
+            }
+        }
+
+        public void StopRunning()
+        {
+            try
+            {
+                ValidateHandle();
+                var selector = LibObjC.GetSelector("stopRunning");
+                LibObjC.SendNoResult(
+                    Handle,
+                    selector);
+            } catch (Exception ex)
+            {
+                Console.WriteLine($"Error stopping AVCaptureSession: {ex.Message}");
             }
             
-            var videoDataOutput = videoDataOutputObj.Handle;
-            
-            LibObjC.SendNoResult(
-                Handle,
-                LibObjC.GetSelector("addOutput:"),
-                videoDataOutput);
+        }
+        
+        protected override void Dispose(bool disposing)
+        {
+            try
+            {
+                /*if (Handle != IntPtr.Zero)
```

Comment:
This should be removed if not needed. If you are keeping it because you have some misgivings about it, please note in your comment what those misgivings are.